packages/nextjs/pages/index.tsx

```diff
@@ -1,24 +1,177 @@
+    }
+  }, [DEXBalloonBalance]);
+
+  const { writeAsync: ethToTokenWrite } = useScaffoldContractWrite({
+    contractName: "DEX",
+    functionName: "ethToToken",
+    value: ethToTokenAmount
+  });
+
+  const { writeAsync: tokenToEthWrite } = useScaffoldContractWrite({
+    contractName: "DEX",
+    functionName: "tokenToEth",
+    args: [ethers.utils.parseEther(tokenToETHAmount || "0")]
+  });
+
+  const {writeAsync: depositWrite} = useScaffoldContractWrite({
+    contractName:"DEX",
+    functionName:"deposit",
+    value: depositAmount
+  });
+
+  const {writeAsync: withdrawWrite} = useScaffoldContractWrite({
+    contractName:"DEX",
+    functionName:"withdraw",
+    args: [ethers.utils.parseEther(withdrawAmount || "0")]
+  });
+
+  const {writeAsync: approveWrite} = useScaffoldContractWrite({
+    contractName:"Balloons",
+    functionName:"approve",
+    args: [approveSpender, ethers.utils.parseEther(approveAmount || "0")]
+  });
+
+  const {data: balanceOfWrite} = useScaffoldContractRead({
+    contractName:"Balloons",
+    functionName:"balanceOf",
+    args: [accountBalanceOf]
+  });
+
+  const {data: contractBalance} = useScaffoldContractRead({
+    contractName:"Balloons",
+    functionName:"balanceOf",
+    args: [DEXInfo?.address]
+  });
+
+  const { data: contractETHBalance } = useBalance({
+    address: DEXInfo?.address,
+  })
+
   return (
     <>
       <MetaHeader />
-      <div className="flex items-center flex-col flex-grow pt-10">
-        <div className="px-5">
-          <h1 className="text-center mb-8">
+      <h1 className="text-center mb-4 mt-9">
             <span className="block text-2xl mb-2">SpeedRunEthereum</span>
-            <span className="block text-4xl font-bold">Challenge #X: Challenge Title </span>
+            <span className="block text-4xl font-bold">Challenge 4: Minimum Viable Exchange </span>
           </h1>
-          <p className="text-center text-lg">
-            Get started by editing{" "}
-            <code className="italic bg-base-300 text-base font-bold">packages/nextjs/pages/index.tsx</code>
-          </p>
-          <p className="text-center text-lg">
-            Edit your smart contract <code className="italic bg-base-300 text-base font-bold">YourContract.sol</code> in{" "}
-            <code className="italic bg-base-300 text-base font-bold">packages/hardhat/contracts</code>
-          </p>
+      <div className="flex items-center flex-col flex-grow pt-10 grid grid-cols-2">
+        <div className="px-5 border rounded-3xl py-5 ml-3">
+              <div className="flex text-center mb-8 px-4 gap-4">
+                <span className="block text-2xl mb-2"><Address address={DEXInfo?.address}/></span>
+                <span><Balance address={DEXInfo?.address}/></span>
+                {isLoading ? (
+                  <span>Loading...</span>
+                ) : (
+                  <span>Balloons: {ethers.utils.formatEther(DEXBalloonBalance?.toString() || 0)}</span>
```

Comment:
<img width="420" alt="image" src="https://github.com/scaffold-eth/se-2-challenges/assets/25638585/f82909cd-3a7b-44f6-969e-f6874b58442e">
where is this icons?)

Reply:
and balloons balance should also be `.toFixed(4)`